FAQ.md

```diff
@@ -211,15 +211,16 @@ this shortest route, you can just stop and avoid wasted effort. In
 sequential land, you might model this "best result" as a shared value
 like `Rc<Cell<usize>>` (here the `usize` represents the length of best
 path found so far); in parallel land, you'd use a `Arc<AtomicUsize>`.
-Now we can make our search function look like:
 
 ```rust
-fn search(path: &Path, cost_so_far: usize, best_cost: &Arc<AtomicUsize>) {
+fn search(path: &Path, cost_so_far: usize, best_cost: &AtomicUsize) {
     if cost_so_far >= best_cost.load(Ordering::SeqCst) {
         return;
     }
     ...
-    best_cost.store(...);
+    // Note: doing `best_cost.store(...)` will introduce race conditions.
+    // Using `fetch_*` such as `fetch_min` will help prevent race conditions.
+    best_cost.fetch_min(cost_so_far, Ordering::SeqCst);
```

Comment:
The idea is that `best_cost` is a total, while `cost_so_far` is only partial. So the first load means "if we're already too big, give up," and then it would finish computing the current cost and eventually try to `fetch_min` that as the new best.

```suggestion
    best_cost.fetch_min(..., Ordering::SeqCst);
```

Or we could expand the first ellipses, something like:

```rust
    let total_cost = cost_so_far + ...;
    best_cost.fetch_min(total_cost, Ordering::SeqCst);
```

Reply:
Went with the first suggestion of changing `fetch_min(..., Ordering::SeqCst)`.